### PR TITLE
feat: add leetcode 5

### PR DESCRIPTION
**Pull request by gbrunofranco**
#### Checklist

- [X] Added description of change
- [X] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Relevant documentation/comments is changed or added
- [X] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [X] Search previous suggestions before making a new one, as yours may be a duplicate.
- [X] I acknowledge that all my contributions will be made under the project's license.

Notes: 

Looking around the leetcode folder no other file has a main function and tests. I'll change that if needed.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/630"></a>

### CODE REVIEW

**Comment by kvedala** [comment]
Please refer: https://github.com/TheAlgorithms/C/pull/571#issuecomment-658248324 and https://github.com/TheAlgorithms/C/pull/621#issuecomment-702163770
Thank you

File: leetcode/src/5.c [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main()
+{
+    test();  // execute the tests
+    // code here
+    return 0;
+}
```
Comment (kvedala) [inline leetcode/src/5.c:92]: Please add a newline character at the end of file.

**Review by kvedala — APPROVED** [review]
👍 Good work. Thank you.
Please add a newline character at the end of the file.

**Comment by gbrunofranco** [comment]
I've moved the file after reading @kvedala's comment in #621 but the check still fails and apparently that's because it's still testing /leetcode/src/5.c and I'm not sure why that is.

**Pull request closed without merging**